File: WebApplication/ClientApp/src/app/_orchestration/getCompetetorsForIrp/getCompetetorsForIrpDto.ts
import { AthleteResultBase } from "../athleteResultBase";

export interface GetCompetetorsForIrpDto {
  fasterAthletes: IrpCompetetorDto[]
  slowerAthletes: IrpCompetetorDto[]
}

export interface IrpCompetetorDto extends AthleteResultBase {
  athleteCourseId: number
  firstName: string
}

File: Orchestration/GetCompetetorsForIrp/GetCompetetorsForIrpDto.cs
namespace Orchestration.GetCompetetorsForIrp;

public class GetCompetetorsForIrpDto
{
	public List<IrpCompetetorDto> FasterAthletes { get; }

	public List<IrpCompetetorDto> SlowerAthletes { get; }

	public GetCompetetorsForIrpDto(List<IrpCompetetorDto> fasterAthletes, List<IrpCompetetorDto> slowerAthletes)
	{
		FasterAthletes = fasterAthletes;
		SlowerAthletes = slowerAthletes;
	}
}
